assets/js/admin/settings-advertise-asc.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package FacebookCommerce
+ */
+jQuery( window ).on( 'load' , function() {
+  const $ = jQuery;
+  const view_items_id_prefix = 'woocommerce-facebook-settings-advertise-asc-';
+  const asc_type_rt = 'retargeting';
+  const asc_type_nb = 'new-buyers';
+
+  let default_values = {};
+  let changes = {};
+  let curr_views = {};
+  curr_views[ asc_type_nb ] = false;
+  curr_views[ asc_type_rt ] = false;
+
+  function read_default_values() {
+
+    default_values[ asc_type_rt ] = read_current_values( asc_type_rt );
+    default_values[ asc_type_nb ] = read_current_values( asc_type_nb );
+
+  }
+
+  function show_busy_indicator( view, type ) {
+
+    $( '#' + view + '-' + type + '-busy-indicator' ).addClass( 'busy-indicator' );
+
+  }
+
+  function hide_busy_indicator( view, type ) {
+
+    $( '#' + view + '-' + type + '-busy-indicator' ).removeClass( 'busy-indicator' );
+
+  }
+
+  function create_modal( link ) {
+    new $.WCBackboneModal.View({
+      target: 'facebook-for-woocommerce-modal',
+      string: {
+        message: '<iframe src="' + link + '" ></iframe>'
+      }
+    });
+  }
+
+  function read_current_values( view ) {
+
+    try {
+      let is_valid = true;
+
+      let prefix = view_items_id_prefix + view + '-';
+      var p1 = $( '#' + prefix + 'p1-input' ).is( ':checked' );
+
+      let element = $( '#' + prefix + 'p2' );
+      element.get( 0 ).setAttribute('aria-invalid', ! element.get( 0 ).checkValidity())
+      is_valid = is_valid && element.get( 0 ).checkValidity();
+      var p2 = element.val();
+
+      element = $( '#' + prefix + 'p4' );
+      element.get( 0 ).setAttribute('aria-invalid', ! element.get( 0 ).checkValidity())
+      is_valid = is_valid && element.get( 0 ).checkValidity();
+      var p4 = element.val();
+
+      element = $( '#' + prefix + 'p5' );
+      element.get( 0 ).setAttribute('aria-invalid', ! element.get( 0 ).checkValidity())
+      is_valid = is_valid && element.get( 0 ).checkValidity();
+      var p5 = element.val();
+
+      return {
+        'valid' : is_valid,
+        'p1': String( p1 ),
+        'p2': String( p2 ),
+        'p3': '',
+        'p4': String( p4 ),
+        'p5': String( p5 )
+      };
+    } catch (error) {
+      // skip
+      return '';
+    }
+
+  }
+
+  function check_for_changes( view ) {
+    hide_error();
```

Comment:
`hide_error` function has a `view` parameter, but it is not passed to the function call here.